### DISCUSSION TITLE
How to direct different job failures to different slack channels

### DISCUSSION THREAD

**Discussion by gnilrets** [dagster-io/dagster#20142] [Q&A] [answered]
I've been using the [make_slack_on_run_failure_sensor](https://docs.dagster.io/_apidocs/libraries/dagster-slack) sensor for a while now.  Currently, any jobs that fail get reported to a single slack channel.  However, as our jobs and teams grow, I'm finding that I want to categorize or tag certain jobs with metadata, and use that to direct failures of jobs to different slack channels.

It looks like there's a `monitored_jobs` option for this method, but I don't know how to select jobs based on certain metadata tags or the absence of a tag (doc links appear to be broken at the moment).  Any suggestions?

**Comment by garethbrickman** [comment, marked as answer]
I don't think `monitored_jobs` natively supports detecting job tags to do this conveniently- may be worth submitting a feature request for that. 

A workaround is using hand-rolled run failure sensors as this will have access to the `dagster_run` object:

If you want to monitor jobs based on tags, you might need to access the dagster_run object within the sensor function and check the tags manually. Here's a conceptual example based on the information from the documentation and GitHub discussions:

```python
@run_failure_sensor()
def my_run_failure_sensor(context):
    if 'my_tag' in context.dagster_run.tags and context.dagster_run.tags['my_tag'] == 'my_value':
        # Logic to send a Slack alert
        send_slack_alert()
```

In this example, the sensor function checks the tags of the dagster_run object to determine whether to send a Slack alert. This is not using monitored_jobs directly but achieves a similar result by filtering runs based on tags within the sensor function.

**Reply by gnilrets** [reply]
Thanks, this workaround was helpful.
